### ISSUE TITLE
h2_sockpair+trace_test.exe max_concurrent_streams

### ISSUE THREAD

**Issue by dgquintas** [grpc/grpc#12435]
https://grpc-testing.appspot.com/job/gRPC_pr_win/7574/testReport/junit/(root)/c_windows_opt_native/cmake_build_Release_h2_sockpair_trace_test_exe_max_concurrent_streams/

```
...
D0907 05:45:52.133000000  1972 combiner.c:239] C:0078AE38 maybe_finish_one n=0077FEB8
D0907 05:45:52.133000000  1972 combiner.c:284] C:0078AE38 finish old_state=3
I0907 05:45:52.133000000  1972 completion_queue.c:937] RETURN_EVENT[0078EE90]: OP_COMPLETE: tag:00000192 OK
E0907 05:45:52.133000000  1972 cq_verifier.c:273] cq returned unexpected event: OP_COMPLETE: tag:00000192 OK
E0907 05:45:52.133000000  1972 cq_verifier.c:280] expected tags:
tag(102) GRPC_OP_COMPLETE success=1 C:\jenkins\workspace\gRPC_pr_win\workspace_c_windows_opt_native\test\core\end2end\tests\max_concurrent_streams.c:378

```

**Comment by yashykt** [comment]
This looks same as #12201

**Comment by yashykt** [comment]
Fixed in #12463
